File: common/src/main/java/dev/chasem/cobblemonextras/commands/PokeKill.java
package dev.chasem.cobblemonextras.commands;

import com.cobblemon.mod.common.Cobblemon;
import com.cobblemon.mod.common.CobblemonEntities;
import com.cobblemon.mod.common.api.storage.NoPokemonStoreException;
import com.cobblemon.mod.common.api.storage.pc.PCStore;
import com.cobblemon.mod.common.api.storage.pc.link.PCLink;
import com.cobblemon.mod.common.api.storage.pc.link.PCLinkManager;
import com.cobblemon.mod.common.entity.pokemon.PokemonEntity;
import com.cobblemon.mod.common.net.messages.client.storage.pc.OpenPCPacket;
import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.context.CommandContext;
import dev.chasem.cobblemonextras.CobblemonExtras;
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions;
import net.minecraft.entity.Entity;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.server.world.ServerWorld;
import net.minecraft.text.Text;

import java.util.List;

import static net.minecraft.server.command.CommandManager.literal;

public class PokeKill {

    public void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(
                literal("pokekill")
                        .requires(src -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.POKEKILL_PERMISSION))
                            .executes(this::execute)
        );
    }

    private int execute(CommandContext<ServerCommandSource> ctx) {
        ServerWorld serverWorld = ctx.getSource().getWorld();
        final int count = 0;
        // Get all entities in world.
        List<? extends PokemonEntity> allPokemon = serverWorld.getEntitiesByType(CobblemonEntities.POKEMON, pokemonEntity -> true);
        ctx.getSource().sendMessage(Text.of("Killing " + allPokemon.size() + " Pokemon"));
        allPokemon.forEach(pokemonEntity -> {
            // Kill all entities.
            pokemonEntity.setRemoved(Entity.RemovalReason.DISCARDED);
            pokemonEntity.kill();
        });
        return 1;
    }
}

File: common/src/main/kotlin/dev/chasem/cobblemonextras/commands/PokeKill.kt
package dev.chasem.cobblemonextras.commands

import com.cobblemon.mod.common.entity.pokemon.PokemonEntity
import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import dev.chasem.cobblemonextras.CobblemonExtras
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands
import net.minecraft.network.chat.Component
import net.minecraft.world.entity.Entity
import net.minecraft.world.level.entity.EntityTypeTest

class PokeKill {
    fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
                Commands.literal("pokekill")
                        .requires { src: CommandSourceStack? -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.POKEKILL_PERMISSION) }
                        .executes { ctx: CommandContext<CommandSourceStack> -> this.execute(ctx) }
        )
    }

    private fun execute(ctx: CommandContext<CommandSourceStack>): Int {
        val serverWorld = ctx.getSource().server.overworld();
        val count = 0
        // Get all entities in world.
        val allPokemon = serverWorld.getEntities(EntityTypeTest.forClass(PokemonEntity::class.java), { pokemonEntity -> true });
        ctx.getSource().sendSystemMessage(Component.literal("Killing " + allPokemon.size + " Pokemon"))

        allPokemon.forEach { pokemonEntity: PokemonEntity ->
            if (!pokemonEntity.isBattling && pokemonEntity.owner == null) {
                pokemonEntity.setRemoved(Entity.RemovalReason.DISCARDED)
                pokemonEntity.kill()
            }
        }
        return 1
    }
}
